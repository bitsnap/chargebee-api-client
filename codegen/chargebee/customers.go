package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListCustomers generates chargebee client code to fetch all customers
//
// API: https://{site}.chargebee.com/api/v2/customers
//
// Documentation: https://apidocs.chargebee.com/docs/api/customers?lang=curl#list_customers
func GenerateListCustomers() string {
	return client.GenerateList("Customers", "Customer", "chargebee.com/api/v2/customers")
}

// GenerateRetrieveCustomer generates chargebee client code to retrieve specific customer
//
// API: https://{site}.chargebee.com/api/v2/customers/{customer_id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/customers?lang=curl#retrieve_a_customer
func GenerateRetrieveCustomer() string {
	return client.GenerateRetrieve("Customer", "chargebee.com/api/v2/customers")
}

// GenerateCustomersModel generates chargebee Customers domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/customers?lang=curl#customer_attributes
func GenerateCustomersModel() string {
	return client.GenerateModel("Customer", "Customer attributes", "https://apidocs.chargebee.com/docs/api/customers")
}
