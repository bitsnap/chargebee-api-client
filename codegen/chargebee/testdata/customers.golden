# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee customers code generation when generating Chargebee API client should be able to generate customers models" = '''
(string) (len=7120) "type Customer struct {\n    Id string `json:\"id\" validate:\"required\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Phone string `json:\"phone\"`\n    Company string `json:\"company\"`\n    VatNumber string `json:\"vat_number\"`\n    AutoCollection enums.AutoCollectionEnum `json:\"auto_collection\" validate:\"required\"`\n    OfflinePaymentMethod enums.OfflinePaymentMethodEnum `json:\"offline_payment_method\"`\n    NetTermDays int `json:\"net_term_days\" validate:\"required\"`\n    VatNumberValidatedTime *time.Time `json:\"vat_number_validated_time\"`\n    VatNumberStatus enums.VatNumberStatusEnum `json:\"vat_number_status\"`\n    AllowDirectDebit bool `json:\"allow_direct_debit\" validate:\"required\"`\n    IsLocationValid bool `json:\"is_location_valid\"`\n    CreatedAt *time.Time `json:\"created_at\" validate:\"required\"`\n    CreatedFromIp string `json:\"created_from_ip\"`\n    Taxability enums.TaxabilityEnum `json:\"taxability\"`\n    EntityCode enums.EntityCodeEnum `json:\"entity_code\"`\n    ExemptNumber string `json:\"exempt_number\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    Locale string `json:\"locale\"`\n    BillingDate int `json:\"billing_date\"`\n    BillingMonth int `json:\"billing_month\"`\n    BillingDateMode enums.BillingDateModeEnum `json:\"billing_date_mode\"`\n    BillingDayOfWeek enums.BillingDayOfWeekEnum `json:\"billing_day_of_week\"`\n    BillingDayOfWeekMode enums.BillingDayOfWeekModeEnum `json:\"billing_day_of_week_mode\"`\n    PiiCleared enums.PiiClearedEnum `json:\"pii_cleared\"`\n    AutoCloseInvoices bool `json:\"auto_close_invoices\"`\n    Channel enums.ChannelEnum `json:\"channel\"`\n    ActiveId string `json:\"active_id\"`\n    FraudFlag enums.FraudFlagEnum `json:\"fraud_flag\"`\n    PrimaryPaymentSourceId string `json:\"primary_payment_source_id\"`\n    BackupPaymentSourceId string `json:\"backup_payment_source_id\"`\n    InvoiceNotes string `json:\"invoice_notes\"`\n    BusinessEntityId string `json:\"business_entity_id\"`\n    PreferredCurrencyCode string `json:\"preferred_currency_code\"`\n    PromotionalCredits uint64 `json:\"promotional_credits\" validate:\"required\"`\n    UnbilledCharges uint64 `json:\"unbilled_charges\" validate:\"required\"`\n    RefundableCredits uint64 `json:\"refundable_credits\" validate:\"required\"`\n    ExcessPayments uint64 `json:\"excess_payments\" validate:\"required\"`\n    IsEinvoiceEnabled bool `json:\"is_einvoice_enabled\"`\n    EinvoicingMethod enums.EinvoicingMethodEnum `json:\"einvoicing_method\"`\n    Deleted bool `json:\"deleted\" validate:\"required\"`\n    RegisteredForGst bool `json:\"registered_for_gst\"`\n    ConsolidatedInvoicing bool `json:\"consolidated_invoicing\"`\n    CustomerType enums.CustomerTypeEnum `json:\"customer_type\"`\n    BusinessCustomerWithoutVatNumber bool `json:\"business_customer_without_vat_number\"`\n    ClientProfileId string `json:\"client_profile_id\"`\n    UseDefaultHierarchySettings bool `json:\"use_default_hierarchy_settings\"`\n    VatNumberPrefix string `json:\"vat_number_prefix\"`\n    EntityIdentifierScheme string `json:\"entity_identifier_scheme\"`\n    EntityIdentifierStandard string `json:\"entity_identifier_standard\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n    ReferralUrls []ReferralUrl `json:\"referral_urls\"`\n    ExternalCustomerId string `json:\"external_customer_id\"`\n    ReferralSharingUrl string `json:\"referral_sharing_url\" validate:\"required\"`\n    CreatedAt *time.Time `json:\"created_at\" validate:\"required\"`\n    UpdatedAt *time.Time `json:\"updated_at\" validate:\"required\"`\n    ReferralCampaignId string `json:\"referral_campaign_id\" validate:\"required\"`\n    ReferralAccountId string `json:\"referral_account_id\" validate:\"required\"`\n    ReferralExternalCampaignId string `json:\"referral_external_campaign_id\"`\n    ReferralSystem enums.ReferralSystemEnum `json:\"referral_system\" validate:\"required\"`\n    Contacts []Contact `json:\"contacts\"`\n    Id string `json:\"id\" validate:\"required\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\" validate:\"required\"`\n    Phone string `json:\"phone\"`\n    Label string `json:\"label\"`\n    Enabled bool `json:\"enabled\" validate:\"required\"`\n    SendAccountEmail bool `json:\"send_account_email\" validate:\"required\"`\n    SendBillingEmail bool `json:\"send_billing_email\" validate:\"required\"`\n    Type enums.TypeEnum `json:\"type\" validate:\"required\"`\n    Gateway enums.GatewayEnum `json:\"gateway\" validate:\"required\"`\n    GatewayAccountId string `json:\"gateway_account_id\"`\n    Status enums.StatusEnum `json:\"status\" validate:\"required\"`\n    ReferenceId string `json:\"reference_id\" validate:\"required\"`\n    Balances []CustomerBalance `json:\"balances\"`\n    PromotionalCredits uint64 `json:\"promotional_credits\" validate:\"required\"`\n    ExcessPayments uint64 `json:\"excess_payments\" validate:\"required\"`\n    RefundableCredits uint64 `json:\"refundable_credits\" validate:\"required\"`\n    UnbilledCharges uint64 `json:\"unbilled_charges\" validate:\"required\"`\n    CurrencyCode string `json:\"currency_code\" validate:\"required\"`\n    EntityIdentifiers []EntityIdentifier `json:\"entity_identifiers\"`\n    Id string `json:\"id\" validate:\"required\"`\n    Value string `json:\"value\"`\n    Scheme string `json:\"scheme\" validate:\"required\"`\n    Standard string `json:\"standard\"`\n    TaxProvidersFields []TaxProvidersField `json:\"tax_providers_fields\"`\n    ProviderName string `json:\"provider_name\" validate:\"required\"`\n    FieldId string `json:\"field_id\" validate:\"required\"`\n    FieldValue string `json:\"field_value\" validate:\"required\"`\n    ParentId string `json:\"parent_id\"`\n    PaymentOwnerId string `json:\"payment_owner_id\" validate:\"required\"`\n    InvoiceOwnerId string `json:\"invoice_owner_id\" validate:\"required\"`\n    PortalEditChildSubscriptions enums.PortalEditChildSubscriptionsEnum `json:\"portal_edit_child_subscriptions\"`\n    PortalDownloadChildInvoices enums.PortalDownloadChildInvoicesEnum `json:\"portal_download_child_invoices\"`\n    SendSubscriptionEmails bool `json:\"send_subscription_emails\" validate:\"required\"`\n    SendInvoiceEmails bool `json:\"send_invoice_emails\" validate:\"required\"`\n    SendPaymentEmails bool `json:\"send_payment_emails\" validate:\"required\"`\n    PortalEditSubscriptions enums.PortalEditSubscriptionsEnum `json:\"portal_edit_subscriptions\"`\n    PortalDownloadInvoices enums.PortalDownloadInvoicesEnum `json:\"portal_download_invoices\"`\n    SendSubscriptionEmails bool `json:\"send_subscription_emails\" validate:\"required\"`\n    SendInvoiceEmails bool `json:\"send_invoice_emails\" validate:\"required\"`\n    SendPaymentEmails bool `json:\"send_payment_emails\" validate:\"required\"`\n}"
'''
